lib/mayaUsd/ufe/UsdUIInfoHandler.cpp

```diff
@@ -76,13 +77,59 @@ namespace ufe {
 UsdUIInfoHandler::UsdUIInfoHandler()
     : Ufe::UIInfoHandler()
 {
+    // Initialize to invalid values.
+    fInvisibleColor[0] = -1.;
+    fInvisibleColor[1] = -1.;
+    fInvisibleColor[2] = -1.;
+
+    // Register a callback to invalidate the invisible color.
+    fColorChangedCallbackId = MEventMessage::addEventCallback(
+        "DisplayRGBColorChanged", onColorChanged, reinterpret_cast<void*>(this));
+
+    // Immediately update the invisible color to get a starting current value.
+    updateInvisibleColor();
 }
 
-UsdUIInfoHandler::~UsdUIInfoHandler() { }
+UsdUIInfoHandler::~UsdUIInfoHandler()
+{
+    // Unregister the callback used to invalidate the invisible color.
+    if (fColorChangedCallbackId)
+        MMessage::removeCallback(fColorChangedCallbackId);
+}
 
 /*static*/
 UsdUIInfoHandler::Ptr UsdUIInfoHandler::create() { return std::make_shared<UsdUIInfoHandler>(); }
 
+void UsdUIInfoHandler::updateInvisibleColor()
+{
+    // Retrieve the invisible color of the outliner.
+    //
+    // We *cannot* intialize it in treeViewCellInfo() because
+    // that function gets called in a paint event and calling
+    // a command in a painting event can cause a recursive paint
+    // event if commands echoing is on, which can corrupt the
+    // Qt paint internal which lead to a crash. Typical symptom
+    // is that the state variable of the Qt paint engine becomes
+    // null midway through the repaint.
+
+    MDoubleArray color;
+    MGlobal::executeCommand("displayRGBColor -q \"outlinerInvisibleColor\"", color);
+
+    if (color.length() == 3) {
```

Comment:
Is this code correct?  color is an std::array, so its length will be 3 on construction.  Perhaps you wanted to check for the uninitialized value, which is -1?

Reply:
Ah, I see, confused fInvisibleColor with color, looks good then.